app/code/Magento/Inventory/Model/ResourceModel/Stock/StockItemQuantity.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\ResourceModel\Stock;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Indexer\Alias;
+use Magento\Inventory\Indexer\IndexNameBuilder;
+use Magento\Inventory\Indexer\IndexNameResolverInterface;
+use Magento\Inventory\Indexer\StockItem\IndexStructure as StockItemIndex;
+use Magento\Inventory\Indexer\StockItemIndexerInterface;
+
+/**
+ * The resource model responsible for retrieving StockItem Quantity.
+ * Used by Service Contracts that are agnostic to the Data Access Layer.
+ */
+class StockItemQuantity
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var IndexNameBuilder
+     */
+    private $indexNameBuilder;
+
+    /**
+     * @var IndexNameResolverInterface
+     */
+    private $indexNameResolver;
+
+    /**
+     * StockItemQuantity constructor.
+     *
+     * @param ResourceConnection $resource
+     * @param IndexNameBuilder $indexNameBuilder
+     * @param IndexNameResolverInterface $indexNameResolver
+     */
+    public function __construct(
+        ResourceConnection $resource,
+        IndexNameBuilder $indexNameBuilder,
+        IndexNameResolverInterface $indexNameResolver
+    ) {
+        $this->resource = $resource;
+        $this->indexNameBuilder = $indexNameBuilder;
+        $this->indexNameResolver = $indexNameResolver;
+    }
+
+    /**
+     * Given a product sku and a stock id, return stock item quantity.
+     *
+     * @param string $sku
+     * @param int $stockId
+     * @return float
+     */
+    public function execute(string $sku, int $stockId): float
+    {
+        $indexName = $this->indexNameBuilder
+            ->setIndexId(StockItemIndexerInterface::INDEXER_ID)
+            ->addDimension('stock_', $stockId)
+            ->setAlias(Alias::ALIAS_MAIN)
+            ->create();
```

Comment:
BTW , why do we have a builder with method `create`, but not `build`
@naydav  that's probably question to you